src/pres/MainContainer.js

```diff
@@ -107,6 +108,74 @@ export default class MainContainer extends React.Component {
     return darkModeCookie === 'true';
   }
 
+  getSettingsFromUrlParams() {
+    const params = new URLSearchParams(window.location.search)
+
+    // Step 0: Variant
+    const variant = params.get("variant") || Constants.VARIANT_STANDARD
+
+    // Step 1: Source
+    const site = params.get("source") || ''
+
+    // Step 2: Player details
+    const playerName = params.get("playerName") || ''
+
+    // Step 3: Color and filters
+    const isAdvancedFiltersOpen = params.get("advanced") === 'true'
+    // 'white' | 'black'
+    const color = params.get("color") || ''
+    // 'all' | 'rated' | 'casual' (corresponds to Filters.toggleRated)
+    const ratedMode = params.get("ratedMode") || "all"
+    // See parseTimeControls. Defaults to all time controls.
+    const timeControls = this.parseTimeControls(params.get("timeControls"))
+    // YYYY-MM-DD
+    const fromDate = params.get("fromDate") ? parseDate(params.get("fromDate")) : null
+    // YYYY-MM-DD
+    const toDate = params.get("toDate") ? parseDate(params.get("toDate")) : null
```

Comment:
Another way to indent this (which I personally prefer, but may not match your style -- your call)

```js
const isAdvancedFiltersOpen =
  params.get("advanced") === 'true'
const color = // 'white' | 'black'
  params.get("color") || ''
const ratedMode = // 'all' | 'rated' | 'casual' (corresponds to Filters.toggleRated)
  params.get("ratedMode") || "all"
const timeControls = // See parseTimeControls. Defaults to all time controls.
  this.parseTimeControls(params.get("timeControls"))
const fromDate = // YYYY-MM-DD
  params.get("fromDate") ? parseDate(params.get("fromDate")) : null
const toDate = // YYYY-MM-DD
  params.get("toDate") ? parseDate(params.get("toDate")) : null
```